cardinal/ecs/storage/engine.go

```diff
@@ -4,3 +4,13 @@ type NonceStorage interface {
 	GetNonce(key string) (uint64, error)
 	SetNonce(key string, nonce uint64) error
 }
+
+type SchemaStorage interface {
+	GetSchema(componentName string) ([]byte, error)
+	SetSchema(componentName string, schemaData []byte) error
+}
+
+type SchemaAndNonceStorage interface {
```

Comment:
we should call this either `EngineStorage` so it can be a catch all for any subsequent storage we might want to add